Source/Core/Core/IPC_HLE/WII_IPC_HLE_Device_DI.cpp

```diff
@@ -63,14 +78,17 @@ IPCCommandResult CWII_IPC_HLE_Device_di::IOCtl(u32 _CommandAddress)
 		Memory::Memset(BufferOut, 0, BufferOutSize);
 	}
 
-	DVDCommandResult result = ExecuteCommand(command_0, command_1, command_2,
-	                                         BufferOut, BufferOutSize, false);
-	Memory::Write_U32(result.interrupt_type, _CommandAddress + 0x4);
+	// The original hardware overwrites the command type with the async reply type.
+	Memory::Write_U32(IPC_REP_ASYNC, _CommandAddress);
+	// IOS also seems to write back the command that was responded to in the FD field.
+	Memory::Write_U32(Memory::Read_U32(_CommandAddress), _CommandAddress + 8);
 
-	if (SConfig::GetInstance().m_LocalCoreStartupParameter.bFastDiscSpeed)
-		result.ticks_until_completion = 0;	// An optional hack to speed up loading times
+	// DVDInterface's ExecuteCommand handles timing, disc reading... basically everything
+	ExecuteCommand(command_0, command_1, command_2, BufferOut, BufferOutSize,
+	               false, ioctl_callback, _CommandAddress);
 
-	return { true, result.ticks_until_completion };
+	// The reply is sent by IOCtlCallback instead of WII_IPC_HLE_Interface
+	return IPC_NO_REPLY;
```

Comment:
It isn't complete anarchy: even without last_reply_time, the queues for replies and acknowledgements force some ordering onto requests/replies.  And DVD read requests will come out in the right order because that essentially has its own queue.  Any delay based on accessing some particular device (like the DVD drive or NAND) probably needs to queue requests, but I would guess most commands take very little CPU time on Starlet.

Maybe we could add some sort of rough CPU-time approximation, like saying that Starlet can only generate 1 reply every 250 us or whatever, but it doesn't make sense to freeze the reply queue for hundreds of milliseconds because Starlet is waiting for the DVD drive to generate an interrupt.  The system is clearly designed to allow out-of-order replies.